### PR TITLE
EE-1048: track total supply

### PR DESCRIPTION
**Pull request by EdHastingsCasperLabs**
This PR adds a running tracker of total supply created by the mint.

https://casperlabs.atlassian.net/browse/EE-1048

### CODE REVIEW

**Review by mpapierski — APPROVED** [review]
Other than one nit picky comment regarding lazy initialization of a URef (which could be created in `mint-install` and assumed existing through the code) it's good to go from my side

File: types/src/mint.rs [hunk 1]
```diff
@@ -78,12 +95,19 @@ pub trait Mint: RuntimeProvider + StorageProvider {
 
     /// Retrieves the base round reward.
     fn read_base_round_reward(&mut self) -> Result<U512, Error> {
-        let base_round_reward_uref = match self.get_key(BASE_ROUND_REWARD_KEY) {
+        let total_supply_uref = match self.get_key(TOTAL_SUPPLY_KEY) {
             Some(Key::URef(uref)) => uref,
             Some(_) => return Err(Error::MissingKey), // TODO
             None => return Err(Error::MissingKey),
```
[the remark this answers is not in the document]
Comment (EdHastingsCasperLabs) [reply types/src/mint.rs:101]: No. It is created during genesis when the first genesis account is created and will exist thereafter.

File: grpc/tests/src/test/step.rs [hunk 2]
```diff
@@ -9,21 +9,21 @@ use casper_execution_engine::{
...
         SEIGNIORAGE_RECIPIENTS_SNAPSHOT_KEY, VALIDATOR_REWARD_PURSE,
     },
     ContractHash, Key, ProtocolVersion, PublicKey,
 };
 
 const ACCOUNT_1_PK: PublicKey = PublicKey::Ed25519([200; 32]);
 const ACCOUNT_1_ADDR: AccountHash = AccountHash::new([200; 32]);
-const ACCOUNT_1_BALANCE: u64 = 10_000_000;
-const ACCOUNT_1_BOND: u64 = 100_000;
+const ACCOUNT_1_BALANCE: u64 = 10_000_000_000_000;
```
Comment (henrytill) [inline grpc/tests/src/test/step.rs:20]: Probably don't _need_ to bump these now.  In fact, I'd like to make sure that the only problem before was indeed the `BLOCK_REWARD` one...

**Review by henrytill — COMMENTED** [review]
Optimistically approving.  Could you go ahead and delete `round_reward::initial_supply_motes`?

**Review by henrytill — APPROVED** [review]
Ach, now I'm approving :laughing:

**Review by AlexanderLimonov — APPROVED** [review]
Seems like the tests, about the only thing I understand here, check most of the things they need to check, so, approved.

File: grpc/tests/src/test/step.rs (hunk 2 above)
Comment (EdHastingsCasperLabs) [reply grpc/tests/src/test/step.rs]: did this

**Pull request merged**